lib/elixir/lib/calendar/time.ex

```diff
@@ -98,37 +98,46 @@ defmodule Time do
       {:error, :invalid_time}
       iex> Time.new(23, 59, 60, 999_999)
       {:error, :invalid_time}
+
+      # Invalid microsecond
       iex> Time.new(23, 59, 59, 1_000_000)
-      {:error, :invalid_time}
+      ** (FunctionClauseError) no function clause matching in Time.new/5
 
-      # Invalid precision
-      Time.new(23, 59, 59, {999_999, 10})
-      {:error, :invalid_time}
+      iex> Time.new(23, 59, 59, {999_999, 10})
+      ** (FunctionClauseError) no function clause matching in Time.new/5
 
   """
   @spec new(
           Calendar.hour(),
           Calendar.minute(),
           Calendar.second(),
-          Calendar.microsecond() | integer,
+          Calendar.microsecond() | 0..999_999,
```

Comment:
Please revert. `0..999_999` applies to the Calendar.ISO, it may not apply to others.

Reply:
Bear in mind that `0..999_999` comes from `t:Calendar.microseconds/0` which is what  
`Calendar.ISO` uses.

Reply:
That should be eventually updated then. I will tackle it later.

Reply:
then should this spec be `Calendar.microsecond() | non_neg_integer,` ? instead of `interger()` ?